echoregions/regions2d/regions2d.py

```diff
@@ -358,122 +356,129 @@ def plot(
         else:
             raise TypeError(
-                f"region_ids must be of type list. Currently is of type {type(region_ids)}"
+                f"region_id must be of type list. Currently is of type {type(region_id)}"
             )
 
-        if isinstance(mask_labels, list) and (len(mask_labels) != len(region_ids)):
-            raise ValueError(
-                "If mask_labels is a list, it should be of same length as region_ids."
-            )
-
-        # Replace nan depth in regions2d.
-        self.replace_nan_depth(inplace=True)
-
         # Dataframe containing region information.
-        region_df = self.select_region(region_ids)
+        region_df = self.select_region(region_id)
 
-        # Select only columns which are important.
+        # Select only important columns
         region_df = region_df[["region_id", "time", "depth"]]
 
-        # Organize the regions in a format for region mask.
-        df = region_df.explode(["time", "depth"])
+        # Filter rows with depth values within self min and self max depth
+        region_df = region_df[
+            (
+                (
+                    region_df["depth"].apply(
+                        lambda x: all(self.min_depth <= i <= self.max_depth for i in x)
+                    )
+                )
+                & (region_df["depth"].apply(lambda x: all(i >= 0 for i in x)))
+            )
+        ]
 
-        # Convert region time to integer timestamp.
-        df["time"] = matplotlib.dates.date2num(df["time"])
+        if region_df.empty:
+            print(
+                "Post NaN Depth Filtered Regions is Empty. All rows had NaN Depth values."
+            )
+        else:
+            # Organize the regions in a format for region mask.
+            df = region_df.explode(["time", "depth"])
 
-        # Create a list of dataframes for each regions.
-        grouped = list(df.groupby("region_id"))
+            # Convert region time to integer timestamp.
+            df["time"] = matplotlib.dates.date2num(df["time"])
 
-        # Convert to list of numpy arrays which is an acceptable format to create region mask.
-        regions_np = [np.array(region[["time", "depth"]]) for id, region in grouped]
+            # Create a list of dataframes for each regions.
+            grouped = list(df.groupby("region_id"))
 
-        # Corresponding region ids converted to int.
-        region_ids = [int(id) for id, region in grouped]
+            # Convert to list of numpy arrays which is an acceptable format to create region mask.
+            regions_np = [np.array(region[["time", "depth"]]) for _, region in grouped]
 
-        # Convert ping_time to unix_time since the masking does not work on datetime objects.
-        da_Sv = da_Sv.assign_coords(
-            unix_time=(
-                "ping_time",
-                matplotlib.dates.date2num(da_Sv.coords["ping_time"].values),
-            )
-        )
+            # Corresponding region ids converted to int.
```

Comment:
Just to make comment format consistent
```suggestion
            # Convert corresponding region_id to int.
```